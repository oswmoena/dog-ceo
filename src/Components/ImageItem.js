import React, { useState, useCallback } from 'react'
import { makeStyles, createStyles } from '@mui/styles';
import { ImageList, ImageListItem, ImageListItemBar } from '@mui/material';
import ImageViewer from 'react-simple-image-viewer';

const useStyles = makeStyles((theme) =>
    createStyles({
        root: {
            display: 'flex',
            flexWrap: 'wrap',
            justifyContent: 'space-around',
            overflow: 'hidden',
        },
        imageList: {
            flexWrap: 'nowrap',
            transform: 'translateZ(0)',
        },
        title: {
            color: '#FFFFFF',
        },
        titleBar: {
            background:
                'linear-gradient(to top, rgba(0,0,0,0.7) 0%, rgba(0,0,0,0.3) 70%, rgba(0,0,0,0) 100%)',
        },
    }));

export const ImageItem = ({ images, title }) => {
    const [currentImage, setCurrentImage] = useState(0);
    const [isViewerOpen, setIsViewerOpen] = useState(false);
    const classes = useStyles();

    const openImageViewer = useCallback((index) => {
        setCurrentImage(index);
        setIsViewerOpen(true);
    }, []);

    const closeImageViewer = () => {
        setCurrentImage(0);
        setIsViewerOpen(false);
    };

    return (
        <div className={classes.root}>
            <ImageList className={classes.imageList} cols={3}>
                {images.map((item, index) => (
                    <ImageListItem key={item}>
                        <img
                            src={item}
                            onClick={() => openImageViewer(index)}
                            width="300"
                            key={index}
                            style={{ margin: '2px' }}
                            alt={title}
                        />
                        <ImageListItemBar
                            title={title}
                            classes={{
                                root: classes.titleBar,
                                title: classes.title,
                            }}
                            loading="lazy"
                        />
                    </ImageListItem>
                ))}
            </ImageList>
            {isViewerOpen && (
                <ImageViewer
                    src={images}
                    currentIndex={currentImage}
                    disableScroll={false}
                    closeOnClickOutside={true}
                    onClose={closeImageViewer}
                />
            )}
        </div>
    )




}
