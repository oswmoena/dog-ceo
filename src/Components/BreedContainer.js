import React, { useEffect, useState } from 'react'
import { getBreed } from '../Providers/Provider'
import { ImageItem } from './ImageItem'
import AutorenewIcon from '@mui/icons-material/Autorenew';
import { createStyles, makeStyles } from '@mui/styles';

const useStyles = makeStyles((theme) => createStyles({
    spacing: {
        display: 'flex',
        justifyContent: 'center',
        padding: '5%',
        fontSize: '20px'
    },
    'animatedItem': {
        animation: `$spin 2s infinite linear`,
    },
    '@global': {
        '@keyframes spin': {
            '0%': {
                transform: 'rotate(0deg)',
            },
            '100%': {
                transform: 'rotate(360deg)',
            },
        },
    },
}))

export const BreedContainer = ({ breed }) => {
    const classes = useStyles();
    const [error, setError] = useState(false)
    const [breedImage, setBreedImage] = useState([])

    useEffect(() => {
        getBreedSelected()
    }, []) // eslint-disable-line react-hooks/exhaustive-deps

    const getBreedSelected = async () => {
        const response = await getBreed(breed.toLowerCase());
        if (response) {
            setError(false)
            setBreedImage(response.message)
        } else {
            setError(true)
        }
    }

    if (error) {
        return (
            <div className={classes.spacing}>
                <p>Error en la comunicaci√≥n con el servidor</p>
            </div>
        )
    }

    return breedImage.length > 0
        ? (
            <ImageItem images={breedImage} title={breed} />
        ) : (
            <div className={classes.spacing}>
                <AutorenewIcon className={classes.animatedItem} />
            </div>
        )


}
