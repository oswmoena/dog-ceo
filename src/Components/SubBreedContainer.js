import { makeStyles, createStyles } from '@mui/styles'
import React, { useState, useEffect } from 'react'
import { getSubBreed } from '../Providers/Provider'
import { ImageItem } from './ImageItem'
import AutorenewIcon from '@mui/icons-material/Autorenew';


const useStyles = makeStyles((theme) => createStyles({
    spacing: {
        display: 'flex',
        justifyContent: 'center',
        padding: '5%',
        fontSize: '20px'
    },
}))

export const SubBreedContainer = ({ breed, subbreed }) => {
    const classes = useStyles();
    const [subBreedImage, setSubBreedImage] = useState([])
    const [error, setError] = useState(false)

    useEffect(() => {
        getSubBreedSelected()
    }, []) // eslint-disable-line react-hooks/exhaustive-deps

    const getSubBreedSelected = async () => {
        const response = await getSubBreed(breed.toLowerCase(), subbreed.toLowerCase());
        if (response) {
            setError(false)
            setSubBreedImage(response.message)
        } else {
            setError(true)
        }
    }

    if (error) {
        return (
            <div className={classes.spacing}>
                <p>Error en la comunicaci√≥n con el servidor</p>
            </div>
        )
    }

    return subBreedImage.length > 0
        ? (
            <ImageItem images={subBreedImage} title={subbreed} />
        ) : (
            <div className={classes.spacing}>
                <AutorenewIcon className={classes.animatedItem} />
            </div>
        )
}
