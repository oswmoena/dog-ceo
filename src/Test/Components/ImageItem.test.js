import React from 'react';
import { shallow } from 'enzyme'
import { ImageItem } from '../../Components/ImageItem';


describe('Test on Image Item component', () => {
    test('Show ImageItem correctly', () => {
        const wrapper = shallow(<ImageItem />);
        expect(wrapper).toMatchSnapshot()
    })
})