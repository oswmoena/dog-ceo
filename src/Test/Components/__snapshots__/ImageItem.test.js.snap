// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Image Item component Show ImageItem correctly 1`] = `
<div
  className="makeStyles-root-1"
>
  <ForwardRef(ImageList)
    className="makeStyles-imageList-2"
    cols={3}
  />
</div>
`;
